va/caa-util.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package va
+
+import (
+	"encoding/hex"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/miekg/dns"
+)
+
+// CAA Holds decoded CAA record.
+type CAA struct {
+	flag     uint8
+	tag      string
+	value    string
+	valueBuf []byte
+}
+
+// RFC 6844 based CAA record decoder
+func newCAA(encodedRDATA []byte) *CAA {
+	if len(encodedRDATA) < 2 {
+		// *very* badly formatted record, discard
+		return nil
+	}
+
+	// first octet is uint8 flags
+	flag := uint8(encodedRDATA[0])
+
+	// second octet is uint8 length of tag
+	tagLen := uint8(encodedRDATA[1])
+
+	if uint8(len(encodedRDATA)) < 2+tagLen {
+		// badly formatted record, discard
```

Comment:
It'd be nice if we could check ==, for correctness' sake. I don't know how well-formatted the results are from the resolver though.